src/vector.rs

```diff
@@ -283,6 +298,32 @@ where
     {
         self.square_length().sqrt()
     }
+
+    /// Returns this vector projected onto another one.
+    #[inline]
+    pub fn project_onto(&self, onto: Self) -> Self
+    where
+        T: Div<T, Output = T>
+    {
+        onto * (self.dot(onto) / onto.square_length())
+    }
+}
+
+impl<T, U> Vector2D<T, U>
+where
+    T: Copy + Mul<T, Output = T> + Add<T, Output = T> + Sub<T, Output = T>
+    + PartialOrd + Float
+{
+    /// Return this vector capped to a maximum length.
+    #[inline]
+    pub fn cap_length(&self, max_length: T) -> Self {
+        let square_length = self.square_length();
+        if square_length > max_length * max_length {
+            return self.normalize() * max_length;
```

Comment:
why early return instead of just if/else expression?

Reply:
It's a general personal preference of favoring early exits over nesting/indentation. It doesn't matter much in this case since the function is very simple. Let me know if you care about this and I'll change it.